playbooks/roles/ocp-customization/files/powersmt

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+export PATH=/root/.local/bin:/root/bin:/sbin:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
+export KUBECONFIG=/var/lib/kubelet/kubeconfig
+COREPS=$(/bin/lscpu | /bin/awk -F: ' $1 ~ /^Core\(s\) per socket$/ {print $2}'|/bin/xargs)
+SOCKETS=$(/bin/lscpu | /bin/awk -F: ' $1 ~ /^Socket\(s\)$/ {print $2}'|/bin/xargs)
+let TOTALCORES=$COREPS*$SOCKETS
+MAXTHREADS=$(/bin/lscpu | /bin/awk -F: ' $1 ~ /^CPU\(s\)$/ {print $2}'|/bin/xargs)
+let MAXSMT=$MAXTHREADS/$TOTALCORES
+CURRENTSMT=$(/bin/lscpu | /bin/awk -F: ' $1 ~ /^Thread\(s\) per core$/ {print $2}'|/bin/xargs)
+
+while :
+do
+  SMTLABEL=$(/bin/oc get node $HOSTNAME -L SMT --no-headers |/bin/awk '{print $6}')
+  if [[ -n $SMTLABEL ]]
+    then
+      case $SMTLABEL in
+        1) TARGETSMT=1
+      ;;
+        2) TARGETSMT=2
+      ;;
+        4) TARGETSMT=4
+      ;;
+        8) TARGETSMT=8
+      ;;
+        *) TARGETSMT=$CURRENTSMT ; echo "SMT value must be 1, 2, 4, or 8 and smaller than Maximum SMT."
+      ;;
+      esac
+    else
+      TARGETSMT=$MAXSMT
+  fi
+
+  CURRENTSMT=$(/bin/lscpu | /bin/awk -F: ' $1 ~ /^Thread\(s\) per core$/ {print $2}'|/bin/xargs)
+
+  if [[ $CURRENTSMT -ne $TARGETSMT ]]
+    then
+      INITONTHREAD=0
+      INITOFFTHREAD=$TARGETSMT
+      if [[ $MAXSMT -ge $TARGETSMT ]]
+        then
+          while [[ $INITONTHREAD -lt $MAXTHREADS ]]
+          do
+            ONTHREAD=$INITONTHREAD
+            OFFTHREAD=$INITOFFTHREAD
+
+            while [[ $ONTHREAD -lt $OFFTHREAD ]]
+            do
+              /bin/echo 1 > /sys/devices/system/cpu/cpu$ONTHREAD/online
+              let ONTHREAD=$ONTHREAD+1
+            done
+            let INITONTHREAD=$INITONTHREAD+$MAXSMT
+            while [[ $OFFTHREAD -lt $INITONTHREAD ]]
+            do
+              /bin/echo 0 > /sys/devices/system/cpu/cpu$OFFTHREAD/online
+              let OFFTHREAD=$OFFTHREAD+1
+            done
+            let INITOFFTHREAD=$INITOFFTHREAD+$MAXSMT
+          done
+        systemctl restart kubelet
```

Comment:
Since we are rebooting the node, we should avoid restarting kubelet

Reply:
Btw, is this setting persistent when done in this way?

Reply:
yes it is persistent as it seems to start before kubelet.
Do you think it is really dangerous to restart kubelet ?

Reply:
I'm not sure how MCO will behave if kubelet is restarted. Additionally since anyway node reboot will happen post application of the machine config then why to restart kubelet.

Reply:
MCO deploy a systemd service which start a script that monitor every 30s the status of the SMT label for the current node. 
If this label is set (or changed) the script change the SMT mode on the current node and restart kubelet to update the CPU resource count without rebooting the node (which is REALLY long on powerVM).
So the main idea is to avoid a reboot when changing mode. (and because I’m 90% sure that users will always forget to reboot the node to refresh the kubelet)

Reply:
There is kubelet, there is crio and then there is impact to the containers that are already running. Some scenarios that I can think of and have no idea how will it get handled if kubelet is restarted.
1. number of logical cpus < total cpu requests for PODs
2. number of logical cpus < total cpu limits for PODs

I agree reboot will be longer process. However according to me the safest way to set the SMT value during install time (as a `oneshot` service) via machineconfig and letting MCO take care of the reboot. 

Let's think through this and get some datapoints. The base code as part of this PR is good start, it's just finding the right way and agreeing to it.